core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -782,7 +782,7 @@ object KafkaConfig {
       .define(SocketSendBufferBytesProp, INT, Defaults.SocketSendBufferBytes, HIGH, SocketSendBufferBytesDoc)
       .define(SocketReceiveBufferBytesProp, INT, Defaults.SocketReceiveBufferBytes, HIGH, SocketReceiveBufferBytesDoc)
       .define(SocketRequestMaxBytesProp, INT, Defaults.SocketRequestMaxBytes, atLeast(1), HIGH, SocketRequestMaxBytesDoc)
-      .define(MaxConnectionsPerIpProp, INT, Defaults.MaxConnectionsPerIp, atLeast(1), MEDIUM, MaxConnectionsPerIpDoc)
+      .define(MaxConnectionsPerIpProp, INT, Defaults.MaxConnectionsPerIp, atLeast(0), MEDIUM, MaxConnectionsPerIpDoc)
```

Comment:
Not related to this PR. Can we also update the doc string of MaxConnectionsPerIpOverridesDoc.
Currently it is not clear about the format of the config string. 

some thing like this:
A comma separated list of per-ip or hostname overrides to the default maximum number of connections.  Example value is :  `hostName:100,127.0.0.1:200`.